ci/miri.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+set -ex
+
+# We need Tree Borrows as some of our raw pointer patterns are not
+# compatible with Stacked Borrows.
+export MIRIFLAGS="-Zmiri-tree-borrows"
+
+# One target that sets `mem-unaligned` and one that does not,
+# and a big-endian target.
+TARGETS=(x86_64-unknown-linux-gnu \
+    armv7-unknown-linux-gnueabihf \
+    s390x-unknown-linux-gnu)
```

Comment:
Nit, bash doesn't need `\` within arrays 🎉 

```suggestion
TARGETS=(
    x86_64-unknown-linux-gnu
    armv7-unknown-linux-gnueabihf
    s390x-unknown-linux-gnu
)
```